src/rp2_common/hardware_clocks/clocks.c

```diff
@@ -327,10 +327,14 @@ void clock_gpio_init(uint gpio, uint src, uint div) {
         invalid_params_if(CLOCKS, true);
     }
 
+    // set up div
+    invalid_params_if(CLOCKS, div <= 0);
+    uint udiv = div * (1u << 8u);
```

Comment:
Might make sense to replace the `8u` here with `CLOCKS_CLK_GPOUT0_DIV_INT_LSB` ?